frontend/controller/service-worker.js

```diff
@@ -34,25 +40,8 @@ sbp('sbp/selectors/register', {
     }
 
     try {
-      const swRegistration = await navigator.serviceWorker.register('/assets/js/sw-primary.js', { scope: '/' })
-
-      // This ensures that the 'store-client-id' message is always sent to the
-      // service worker, even if it's not immediately available. Calling `removeEventListener` is fine because presumably the page performing an
-      // update is sending this message. However, the real question is whether
-      // we _need_ to store a client ID (broadcasting should be preferred) and
-      // for instances where a single client ID is needed, this should be
-      // done periodically (e.g., to identify an active window).
-      // TODO: Clarify the use and need for 'store-client-id'. This will become
-      // clearer once implementing notifications.
-      if (swRegistration.active) {
-        swRegistration.active?.postMessage({ type: 'store-client-id' })
-      } else {
-        const handler = () => {
-          navigator.serviceWorker.removeEventListener('controllerchange', handler, false)
-          navigator.serviceWorker.controller.postMessage({ type: 'store-client-id' })
-        }
-        navigator.serviceWorker.addEventListener('controllerchange', handler, false)
-      }
+      // Using hash (#) is possible, but seems to get reset when the SW restarts
+      const swRegistration = await navigator.serviceWorker.register(`/assets/js/sw-primary.js${isPwa() ? '?standalone=1' : ''}`, { scope: '/' })
```

Comment:
Claude, commenting on the `swRegistration.update()` line below (which I can't comment on since it didn't change):

> There's a potential race condition here. The initial update() call happens immediately after registration, but the service worker might not be fully activated yet. Consider adding a small delay or waiting for the "activate" event before checking for updates.